jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/ImageNameTest.java

```diff
@@ -123,6 +123,22 @@ private void verifyData(Object[] data) {
     @Test
     void testIllegalFormat() {
         assertThrows(IllegalArgumentException.class, () -> new ImageName(""));
+
+        // New test for too long repository name
+        String tooLongName = generateTooLongImageName();
+        assertThatIllegalArgumentException()
+            .as("Too long image name should fail")
+            .isThrownBy(() -> new ImageName(tooLongName))
+            .withMessageContaining("Repository name must not be more than 255 characters");
+    }
+
+    private String generateTooLongImageName() {
+        StringBuilder tooLongName = new StringBuilder();
+        int maxLength = 255 + 1; // exceeding the maximum length
+        for (int i = 0; i < maxLength; i++) {
```

Comment:
You can replace this by a prebuilt method from Apache's [StringUtils repeat](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#repeat-java.lang.String-int-)